src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -370,6 +370,14 @@ public static IHtmlContent TimeInputTemplate(IHtmlHelper htmlHelper)
             return GenerateTextBox(htmlHelper, inputType: "time");
         }
 
+        public static IHtmlContent MonthInputTemplate(IHtmlHelper htmlHelper)
+        {
+            // A new HTML5 input type that only will be rendered in Rfc3339 mode
+            htmlHelper.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
```

Comment:
I believe this will have the unintended side-effect of changing the rendering mode to Rfc3339 for any input rendered after rendering an input with type=month. You need to do something like

```csharp
var originalRenderingMode = htmlHelper.Html5DateRenderingMode;
try
{
    htmlHelper.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
    ApplyRfc3339DateFormattingIfNeeded(htmlHelper, "{0:yyyy-MM}");
    var result = GenerateTextBox(htmlHelper, inputType: "month");
}
finally
{
    htmlHelper.Html5DateRenderingMode = originalRenderingMode;
}
```

I'm not sure if its important that we also do
```csharp
catch
{
   htmlHelper.Html5DateRenderingMode = originalRenderingMode;
   throw;
}
```
to restore the state before the exception bubbles up, but I don't think so. /cc: @dougbu as he knows all the details about this.

Reply:
None of the above is needed. Only unique `IHtmlHelper` instances are passed into these methods. In particular, https://github.com/aspnet/Mvc/blob/d9825d1547e51619c0e4d6eba710c1f67172e136/src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/TemplateRenderer.cs#L254-L265 is always called.

Reply:
Forgot to mention: The `htmlHelper.ViewContext` is also unique.